internal/infrastructure/kubernetes/proxy/resource_provider.go

```diff
@@ -224,5 +226,46 @@ func (r *ResourceRender) Deployment() (*appsv1.Deployment, error) {
 		},
 	}
 
+	// omit the deployment replicas if HPA is being set
+	if provider.GetEnvoyProxyKubeProvider().EnvoyHpa != nil {
+		deployment.Spec.Replicas = nil
+	}
+
 	return deployment, nil
 }
+
+func (r *ResourceRender) HorizontalPodAutoscaler() (*autoscalingv2.HorizontalPodAutoscaler, error) {
+	provider := r.infra.GetProxyConfig().GetEnvoyProxyProvider()
+	if provider.Type != egv1a1.ProviderTypeKubernetes {
+		return nil, fmt.Errorf("invalid provider type %v for Kubernetes infra manager", provider.Type)
+	}
+
+	hpaConfig := provider.GetEnvoyProxyKubeProvider().EnvoyHpa
+	if hpaConfig == nil {
+		return nil, nil
+	}
+
+	hpa := &autoscalingv2.HorizontalPodAutoscaler{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "autoscaling/v2",
+			Kind:       "HorizontalPodAutoscaler",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: r.Namespace,
+			Name:      r.Name(),
```

Comment:
i think it's necessary to make `labels` configurable, not so rush in this PR, we can do it in the future

Reply:
Can you tell the scenarios ?

Reply:
no specific scenarios on my mind right now. it's not a strong option, can be considered if someone runs into it in the future.